src/shared-components/button/style.ts

```diff
@@ -138,14 +147,23 @@ function parseTheme(disabled, buttonType, isLoading, buttonColor) {
   }
 }
 
+type BaseButtonStylesTypes = {
+  disabled: boolean;
+  buttonType: ButtonTypeWithAction;
+  buttonColor: string;
+  isLoading?: boolean;
```

Comment:
Should we enforce this prop more strictly (required) on the dev side to encourage better handling of the different button states?

Reply:
Resolving comment--this is potentially options since we're using the same function for LinkButtons too.